export/export_handler.go

```diff
@@ -291,3 +297,42 @@ func (h *Handler) Actions() []string {
 func (h *Handler) ActorID() parade.ActorID {
 	return actorName
 }
+
+// exportCommitHook is a cataloger PostCommit hook for continuous export.
+func (h *Handler) exportCommitHook(ctx context.Context, _ db.Tx, repo, branch string, log *catalog.CommitLog) error {
+	l := logging.Default().
+		WithFields(logging.Fields{"repo": repo, "branch": branch, "message": log.Message, "at": log.CreationDate.String()})
+	isContinuous, err := hasContinuousExport(h.cataloger, repo, branch)
+	if err != nil {
+		// FAIL this commit: if we were meant to export it and did not then in practice
+		// there was no commit.
+		return fmt.Errorf("check continuous export for commit %+v: %w", *log, err)
+	}
+	if !isContinuous {
+		return nil
+	}
+	exportID, err := ExportBranchStart(h.parade, h.cataloger, repo, branch)
+	l.WithField("export_id", exportID).Info("continuous export started")
+	if errors.Is(err, ErrExportInProgress) {
+		err = nil
+	}
+	return err
+}
+
+// exportMergeHook is a cataloger PostMerge hook for continuous export.
+func (h *Handler) exportMergeHook(ctx context.Context, _ db.Tx, repo, branch string, merge *catalog.MergeResult) error {
+	isContinuous, err := hasContinuousExport(h.cataloger, repo, branch)
+	if err != nil {
+		// FAIL this merge: if we were meant to export it and did not then in practice
+		// there was no merge.
+		return fmt.Errorf("check continuous export for merge %+v: %w", *merge, err)
+	}
+	if !isContinuous {
+		return nil
+	}
+	_, err = ExportBranchStart(h.parade, h.cataloger, repo, branch)
+	if errors.Is(err, ErrExportInProgress) {
+		err = nil
+	}
```

Comment:
It might be nice to take it out to a function, given that exportCommitHook does the same

Reply:
Thanks, that's a good idea!  I thought it would be too hard because of logging and error messages, but it was actually easy...